cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkSparkConf.java

```diff
@@ -128,12 +128,13 @@ public class BulkSparkConf implements Serializable
     protected boolean useOpenSsl;
     protected int ringRetryCount;
     protected final Set<String> blockedInstances;
+    protected final DigestTypeOption digestTypeOption;
 
     public BulkSparkConf(SparkConf conf, Map<String, String> options)
     {
         this.conf = conf;
         Optional<Integer> sidecarPortFromOptions = MapUtils.getOptionalInt(options, WriterOptions.SIDECAR_PORT.name(), "sidecar port");
-        this.userProvidedSidecarPort = sidecarPortFromOptions.isPresent() ? sidecarPortFromOptions.get() : getOptionalInt(SIDECAR_PORT).orElse(-1);
+        this.userProvidedSidecarPort = sidecarPortFromOptions.orElseGet(() -> getOptionalInt(SIDECAR_PORT).orElse(-1));
```

Comment:
nit: ternary operator reads better than orElseGet

Reply:
I actually prefer the orElseGet approach here. But I will revert since it's not related to this PR

Reply:
Agree with @frankgh here... I prefer `orElseGet` when using optionals.